src/main/java/org/spongepowered/common/service/scheduler/SchedulerBase.java

```diff
@@ -0,0 +1,260 @@
+        this.taskMap.remove(task.getUniqueId());
+    }
+
+    /**
+     * Process all tasks in the map.
+     */
+    protected final void runTick() {
+        this.preTick();
+        try {
+            for (ScheduledTask task : this.taskMap.values()) {
+                this.processTask(task);
+            }
+            this.postTick();
+        } finally {
+            this.finallyPostTick();
+        }
+    }
+
+    /**
+     * Fired when the scheduler begins to tick, before any tasks are processed.
+     */
+    protected void preTick() {
+    }
+
+    /**
+     * Fired when the scheduler has processed all tasks.
+     */
+    protected void postTick() {
+    }
+
+    /**
+     * Fired after tasks have attempted to be processed, in a finally block to
+     * guarantee execution regardless of any error when processing a task.
+     */
+    protected void finallyPostTick() {
+    }
+
+    /**
+     * Processes the task.
+     *
+     * @param task The task to process
+     */
+    protected void processTask(ScheduledTask task) {
+        // If the task is now slated to be cancelled, we just remove it as if it
+        // no longer exists.
+        if (task.getState() == ScheduledTask.ScheduledTaskState.CANCELED) {
+            this.removeTask(task);
+            return;
+        }
+        long threshold = Long.MAX_VALUE;
+        // Figure out if we start a delayed Task after threshold ticks or, start
+        // it after the interval (period) of the repeating task parameter.
+        if (task.getState() == ScheduledTask.ScheduledTaskState.WAITING) {
+            threshold = task.offset;
+        } else if (task.getState() == ScheduledTask.ScheduledTaskState.RUNNING) {
+            threshold = task.period;
+        }
+        // This moment is 'now'
+        long now = this.getTimestamp();
+        // So, if the current time minus the timestamp of the task is greater
+        // than the delay to wait before starting the task, then start the task.
+        // Repeating tasks get a reset-timestamp each time they are set RUNNING
+        // If the task has a period of 0 (zero) this task will not repeat, and
+        // is removed after we start it.
+        if (threshold <= (now - task.getTimestamp())) {
+            task.setTimestamp(this.getTimestamp());
+            startTask(task);
+            // If task is one time shot, remove it from the map.
+            if (task.period == 0L) {
+                this.removeTask(task);
+            }
+        }
+    }
+
+    /**
+     * Begin the execution of a task. Exceptions are caught and logged.
+     *
+     * @param task The task to start
+     */
+    protected void startTask(ScheduledTask task) {
+        task.setState(ScheduledTask.ScheduledTaskState.RUNNING);
+        try {
+            this.executeTaskRunnable(task.getRunnable());
+        } catch (Exception ex) {
+            Sponge.getLogger().error("The Scheduler tried to run the Task, but the Runnable could not be started.");
```

Comment:
Log just one message here and add the exception directly. Also manage the plugin and task name etc when reporting the error, similar to this for example: https://github.com/SpongePowered/SpongeCommon/blob/master/src/main/java/org/spongepowered/common/event/SpongeEventManager.java#L250